src/app/tests/TestReportScheduler.cpp

```diff
@@ -697,17 +717,44 @@ class TestReportScheduler
         NL_TEST_ASSERT(aSuite, node2->GetSyncTimestamp() == node4->GetMaxTimestamp());
         NL_TEST_ASSERT(aSuite, node3->GetSyncTimestamp() == node1->GetMaxTimestamp());
 
-        ctx.GetIOContext().DriveIOUntil(System::Clock::Milliseconds32(1000),
-                                        [&]() -> bool { return syncScheduler.IsReportableNow(readHandler4); });
         // Increment the mock system time to the max to the time waited
         sTestTimerSynchronizedDelegate.IncrementMockTimestamp(System::Clock::Milliseconds64(1000));
 
-        // Confirm  readHandler 1-2-4 are reportable and handler 3 is not because of it min interval
+        // Confirm  readHandler 1-2-4 are reportable and handler 3 is not because of its min interval
         NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler1));
         NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler2));
         NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler3));
         NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler4));
 
+        syncScheduler.OnReadHandlerDestroyed(readHandler1);
+        syncScheduler.OnReadHandlerDestroyed(readHandler2);
+        syncScheduler.OnReadHandlerDestroyed(readHandler3);
+        syncScheduler.OnReadHandlerDestroyed(readHandler4);
+
+        // Reset all handlers
+        // Test case: Scheduler 1 and 2 are reportable but min2 > max1, they should sync only when possible (min2 = 3, max1 = 2)
+        NL_TEST_ASSERT(aSuite, syncScheduler.GetNumReadHandlers() == 0);
+
+        NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler1->SetMaxReportingInterval(2));
+        NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler1->SetMinReportingIntervalForTests(0));
+        syncScheduler.OnReadHandlerCreated(readHandler1);
+        readHandler1->ForceDirtyState();
```

Comment:
The logic for `ForceDirtyState` and `ClearDirtyState` seems off? Sometimes we clear manually sometimes we don't.

Reply:
this test case is incomplete, we should always clear after forcing, at least in this test.